docs/book/tutorials/kubernetes-on-vsphere-with-kubeadm.md

```diff
@@ -1,58 +1,175 @@
+
+## Recommended Guest Operating System
 
-It is assumed that you have Ubuntu 18.04 LTS VMs set up as a template and cloned from to act as base images for your K8s cluster, if you would like guidance on how to do this, please [see here](https://blah.cloud/kubernetes/creating-an-ubuntu-18-04-lts-cloud-image-for-cloning-on-vmware/)
+VMware recommends that you create a virtual machine template using Guest OS Ubuntu 18.04.1 LTS (Bionic Beaver) 64-bit PC (AMD64) Server. Check it out on [VMware PartnerWeb](http://partnerweb.vmware.com/GOSIG/Ubuntu_18_04_LTS.html). This template is cloned to act as base images for your Kubernetes cluster. For instructions on how to do this, please refer to the guidance provided in [this blog post by Myles Gray of VMware](https://blah.cloud/kubernetes/creating-an-ubuntu-18-04-lts-cloud-image-for-cloning-on-vmware/). Ensure that SSH access is enabled on all nodes. This must be done in order to run commands on both the Kubernetes master and worker nodes in this guide.
 
-In the setup guide below we set up a single master and multiple worker nodes.
+## Virtual Machine Hardware requirements
 
-It is also assumed you have SSH access to all nodes in order to run the commands on in the following guide.
+Virtual Machine Hardware must be version 15 or higher. For Virtual Machine CPU and Memory requirements, size adequately based on workload requirements.
+VMware also recommend that virtual machines use the VMware Paravirtual SCSI controller for Primary Disk on the Node VM. This should be the default, but it is always good practice to check.
+Finally, the disk.EnableUUID parameter must be set for each node VMs. This step is necessary so that the VMDK always presents a consistent UUID to the VM, thus allowing the disk to be mounted properly.
+It is recommended to not take snapshots of CNS node VMs to avoid errors and unpredictable behavior.
 
-### Tools
+## Docker Images
 
-We are using macOS here, so the `brew` package manager is used to install and manage the tools, if you are using Linux or Windows, use the appropriate install guide for each tool, according to your OS.
+The following is the list of docker images that are required for the installation of CSI and CPI on Kubernetes. These images are automatically pulled in when CSI and CPI manifests are deployed.
+VMware distributes and recommends the following images:
+
+```bash
+vmware/vsphere-block-csi-driver:v1.0.0
+vmware/volume-metadata-syncer:v1.0.0
+http://gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.0.0
+```
+
+In addition, you can use the following images or any of the open source or commercially available container images appropriate for the CSI deployment:
```

Comment:
Can we remove these docker images listing?
For each different version of Kubernetes, we need different docker images for these components.

Reply:
I think it's worth keeping these since it points to where the images are hosted but I think we should at least have a note mentioning that the tags reference the version of various components in-case it wasn't already clear.

Reply:
Done!